### PR TITLE
IR improvements

### PR DESCRIPTION
**Pull request by Trass3r**
A few small improvements. Shouldn't be squashed.

### CODE REVIEW

File: X86/X86MachineInstructionRaiserUtils.cpp [hunk 1]
```diff
@@ -1518,7 +1518,7 @@ X86MachineInstructionRaiser::getMemoryAddressExprValue(const MachineInstr &MI) {
       Type *MulValTy = IndexRegVal->getType();
       Value *ScaleAmtValue = ConstantInt::get(MulValTy, ScaleAmt);
       Instruction *MulInst =
-          BinaryOperator::CreateMul(ScaleAmtValue, IndexRegVal);
+          BinaryOperator::CreateMul(ScaleAmtValue, IndexRegVal, "memref idxreg");
```
Comment (aaronsm) [inline X86/X86MachineInstructionRaiserUtils.cpp]: Looks like you wanted a dash here or no dashes below

**Comment by Trass3r** [comment]
rebased.

File: X86/X86RaisedValueTracker.cpp [hunk 2]
```diff
@@ -471,21 +474,20 @@ bool X86RaisedValueTracker::testAndSetEflagSSAValue(unsigned int FlagBit,
...
     // Create the instruction
     //      and SubInst, ShiftLeft
-    Instruction *AndInst = BinaryOperator::CreateAnd(ShiftLeft, TestResultVal);
+    Instruction *AndInst = BinaryOperator::CreateAnd(ShiftLeft, TestResultVal, "highbit");
     RaisedBB->getInstList().push_back(AndInst);
     // Compare result of logical and operation to find if bit 31 is set SF
     // accordingly
     Instruction *SFTest =
-        new ICmpInst(CmpInst::Predicate::ICMP_EQ, AndInst, ShiftLeft,
+        new ICmpInst(CmpInst::Predicate::ICMP_NE, AndInst, ZeroVal,
```
Comment (bharadwajy) [inline X86/X86RaisedValueTracker.cpp]: I see the following build error with this change
```
<repo-dir>/llvm/tools/llvm-mctoll/X86/X86RaisedValueTracker.cpp:490:60: error: use of undeclared identifier 'ZeroVal'
        new ICmpInst(CmpInst::Predicate::ICMP_NE, AndInst, ZeroVal,
                                                           ^
1 error generated.
```

File: X86/X86MachineInstructionRaiser.cpp [hunk 3]
```diff
@@ -1295,15 +1295,16 @@ bool X86MachineInstructionRaiser::raiseBinaryOpMemToRegInstr(
...
-    LoadInst *LdInst =
-        new LoadInst(dyn_cast<LoadInst>(MemRefValue)->getPointerOperand());
-    LdInst->setAlignment(MaybeAlign(MemAlignment));
-    LoadValue = getRaisedValues()->setInstMetadataRODataContent(LdInst);
-  } else {
-    LoadInst *LdInst = new LoadInst(MemRefValue);
-    LdInst->setAlignment(MaybeAlign(MemAlignment));
-    LoadValue = getRaisedValues()->setInstMetadataRODataContent(LdInst);
+    LoadValue =
+        new LoadInst(dyn_cast<LoadInst>(MemRefValue)->getPointerOperand(),
```
Comment (bharadwajy) [inline X86/X86MachineInstructionRaiser.cpp]: I see the following build error with this change
```
<repo-dir>/llvm/tools/llvm-mctoll/X86/X86MachineInstructionRaiser.cpp:1299:13: error: no matching constructor for initialization of 'llvm::LoadInst'
        new LoadInst(dyn_cast<LoadInst>(MemRefValue)->getPointerOperand(),
            ^        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/bharadwaj/github/src/llvm-project/llvm/include/llvm/IR/Instructions.h:206:3: note: candidate constructor not viable: no known conversion from 'unsigned int' to 'llvm::Instruction *' for 4th argument
  LoadInst(Value *Ptr, const Twine &NameStr, bool isVolatile,
...

```

**Pull request merged**